File: Python/Searching/Linear-Search.py
#Prints only first occurence index of searched value
def LinearSearch(arr,value):
        for i in range(len(arr)):
            if value == arr[i]:
                return i
        return None

#Prints all occurence index of searched value
def LinearSearch1(arr,key): 
    flag = 0  
    for i in range(len(arr)):
        if value == arr[i]:
            print(value,"found at index",i)
            flag = 1
    if flag == 0:
        print("Not Found")

def rec_LinearSearch(arr,start,end,key):
    if start<=end:
        if arr[start] == key:
            return start
        elif arr[end] == key:
            return end
        else:
            return rec_LinearSearch(arr,start+1,end-1,key)

    return -1

array = list(map(int,input("Enter your array (as space separated integers) : ").split()))
value = int(input("Enter the value to be searched : "))

LinearSearch1(array,value)
# print(rec_LinearSearch(array,0,len(array)-1,value))

File: C++/Searching/Linear-Search.cpp
//Including header file for using common functionalities
#include <iostream>

//Declaring function defination, to let 
//compiler know that this function do exists
void Linear_Search(int [], int , int );

int main(){

    int arr[50];
    int n, key;

    std::cout << "Enter the size of your array : " << std::endl;
    std::cin >> n;
    std::cout << "Enter your array elements : " << std::endl;

    //Inputting array
    for(int i=0; i < n; ++i)
        std::cin >>  arr[i];

    std::cout << "Enter value to search : ";
    std::cin >> key;

    Linear_Search(arr,n,key);

    return 0;
}

//Body of the declared function
void Linear_Search(int arr[], int size, int key){

    int flag = 0;

    for(int i=0; i<size; ++i)
        if (arr[i] == key){
            std::cout<< key << " found at index " << i << std::endl;
            flag = 1;
        }

    if(flag == 0)
        std::cout << key << " not found !" << std::endl;    
}